src/retransmit_stage.rs

```diff
@@ -85,19 +87,32 @@ fn retransmit(
             .add_field("count", influxdb::Value::Integer(dq.len() as i64))
             .to_owned(),
     );
-    let retransmit_peers = compute_retransmit_peers(
+    let (neighbors, children) = compute_retransmit_peers(
         &bank,
         cluster_info,
         DATA_PLANE_FANOUT,
         NEIGHBORHOOD_SIZE,
         GROW_LAYER_CAPACITY,
     );
-    for b in &mut dq {
-        ClusterInfo::retransmit_to(&cluster_info, &retransmit_peers, b, sock)?;
+    for b in &dq {
+        if b.read().unwrap().should_forward() {
+            ClusterInfo::retransmit_to(&cluster_info, &neighbors, &for_neighbors(b), sock)?;
+        }
+        // Always send blobs to children
+        ClusterInfo::retransmit_to(&cluster_info, &children, b, sock)?;
     }
     Ok(())
 }
 
+/// Modifies a blob for neighbors nodes
+#[inline]
+fn for_neighbors(b: &SharedBlob) -> SharedBlob {
```

Comment:
Needs a better name

Reply:
Honestly, I could do without this function altogether. I only added it to make sure this diff had good coverage. 
How about `copy_for_neightbors` ?